src/core/public/ui_settings/ui_settings_api.ts

```diff
@@ -65,16 +70,18 @@ export class UiSettingsApi {
    * already in progress it will wait until the previous request is complete
    * before sending the next request
    */
-  public batchSet(key: string, value: any) {
+  public batchSet(key: string, value: any, scope?: UiSettingScope) {
```

Comment:
With some of the regressions related to putting conditional logic within existing stable code that hasnt changed but is shared across multiple plugins. Would it be safer to implement just new functions.

Like `batchSetWithScope`